tp_shared.gpr

```diff
@@ -41,6 +41,26 @@ abstract project TP_Shared is
    Adaflags := external_as_list ("ADAFLAGS", " ");
    Ldflags := external_as_list ("LDFLAGS", " ");
 
+   type OS_Type is ("Linux", "Windows", "macOS", "auto");
+   Target_OS : OS_Type := external ("Target_OS", "auto");
```

Comment:
We already have a variable TARGET defined, so it should be reused.

Also note that the pattern can be far more complex that that, on my side the triplet is x86_64-linux-gnu.

For windows we can have w64 or w32 instead of pc or even mingw32... etc.

But Library_Version is not supported on Windows anyway. We we just need to check if Linux, MacOS or others maybe.

See how PRJ_TARGET has been handled in AWS for this.

Reply:
This is how `gnatcoll` does it:
https://github.com/AdaCore/gnatcoll-core/blob/e398d07e6afa1ef494bbc8784e8ef1d5b856d9a7/gnatcoll.gpr#L74

It requires a `GNATCOLL_OS` external set by the Makefile.

Reply:
@TurboGit, the defined TARGET value comes from gcc -dumpmachine which is quite similar to Project'Target. I brought just few values in the case statement as an example, other user needs may be added thus.
@Fabien-Chouteau, the gprbuild tool is aware of the targeted OS which can be used with Project'Target, shouldn't be this a more reliable usage inside GPR project file?